src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/NativeMemoryTests.cs

```diff
@@ -0,0 +1,383 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public unsafe class NativeMemoryTests
+    {
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(4)]
+        [InlineData(8)]
+        [InlineData(16)]
+        [InlineData(32)]
+        [InlineData(64)]
+        [InlineData(128)]
+        [InlineData(256)]
+        [InlineData(512)]
+        [InlineData(1024)]
+        [InlineData(2048)]
+        [InlineData(4096)]
+        [InlineData(8192)]
+        [InlineData(16384)]
```

Comment:
Any particular reason why to end with `16384`? 

For example, I think that this API may be used with 64k alignment in some cases since that's the page size on some Unix systems.

I do not think we need to cover every power of two starting with 1, but I think we should cover broader range, e.g. `1, 2, 4, 8, 16, 32, 64, 256, 1024, 4096, 65536, 1024*1024`.

Reply:
I just went for reasonable coverage of correctness without being overly verbose. At some point, we just need to trust that the native API is going to do the right thing here.

Reply:
I agree that we have to trust the native API. At the same time, we should sanity check that the implementation of the native API does not have arbitrary limitations that the libraries using this API may run into. It was my reason behind suggesting to add `65536` and `1024*1024`.

Reply:
I'll add covereage for 65k and 1/2/4mb (as those are are also common due to being possible OS page sizes).